test/e2e/gc_e2e_test.go

```diff
@@ -284,4 +284,101 @@ var _ = Describe("Garbage collection for dependent resources", func() {
 
 	})
 
+	Context("When installing a bundle with configmap and secret objects these objects should be deleted when the CSV is deleted", func() {
+		const (
+			sourceName    = "test-catalog"
+			imageName     = "quay.io/olmtest/single-bundle-index:objects"
+			packageName   = "busybox"
+			channelName   = "alpha"
+			subName       = "test-subscription"
+			csvName       = "busybox.v2.0.0"
+			secretName    = "mysecret"
+			configmapName = "special-config"
+		)
+
+		var (
+			source *v1alpha1.CatalogSource
+			err    error
+		)
+
+		BeforeEach(func() {
+			// create catalog source
+			source = &v1alpha1.CatalogSource{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       v1alpha1.CatalogSourceKind,
+					APIVersion: v1alpha1.CatalogSourceCRDAPIVersion,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      sourceName,
+					Namespace: testNamespace,
+					Labels:    map[string]string{"olm.catalogSource": sourceName},
+				},
+				Spec: v1alpha1.CatalogSourceSpec{
+					SourceType: v1alpha1.SourceTypeGrpc,
+					Image:      imageName,
+				},
+			}
+
+			source, err = operatorClient.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.TODO(), source, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred(), "could not create catalog source")
+
+			// Create a Subscription for package
+			_ = createSubscriptionForCatalog(operatorClient, source.GetNamespace(), subName, source.GetName(), packageName, channelName, "", v1alpha1.ApprovalAutomatic)
+
+			// Wait for the Subscription to succeed
+			Eventually(func() error {
+				_, err = fetchSubscription(operatorClient, testNamespace, subName, subscriptionStateAtLatestChecker)
+				return err
+			}).Should(BeNil())
+
+			// confirm extra bundle objects (secret and configmap) are installed
```

Comment:
Since you are verifying that your `CatalogSource` and `Subscription`  which is part of your _pre-setup_ created `secret` & `configMap`, it could be part of its own `It` spec.  For example, your test could look something like: 
```
When("a catalog source is created using grpc and a subscription", func() {
         BeforeEach(func(){
         // create catsrc and subscription  
         })
         It("should have created a secret and a configmap", func() {
                 // validate these objects
         })
```